### PR TITLE
hpcx-mpi: new package

### PR DESCRIPTION
**Pull request by mwkrentel**
This is a virtual package for Nvidia's HPC-X MPI implementation for
external specs only.

----------

I found hpcx mpi on the hpcadvisorycouncil.com machine.

### CODE REVIEW

**Comment by mwkrentel** [comment]
@becker33 @adamjstewart @tldahlgren 
Could I get someone to review?

HPC-X is the vendor compiler from nvidia on the hpcadvisorycouncil.com
machine, derived from openmpi.

I discussed this briefly with @becker33 two months ago and we agreed
that it was cleaner to make a separate package rather than hack the
openmpi recipe, due to the different treatment of the OPAL environ
vars (for starters).

Btw, it seems that some ci test is stuck.

**Comment by mwkrentel** [comment]
It says "ci Pipeline failed," but I can't find any error message.
I guess we'll try the close/open again to see if we have better luck.

File: var/spack/repos/builtin/packages/hpcx-mpi/package.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        self.spec.mpifc  = os.path.join(self.prefix.bin, 'mpif90')
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('MPICC',  os.path.join(self.prefix.bin, 'mpicc'))
+        env.set('MPICXX', os.path.join(self.prefix.bin, 'mpic++'))
+        env.set('MPIF77', os.path.join(self.prefix.bin, 'mpif77'))
+        env.set('MPIF90', os.path.join(self.prefix.bin, 'mpif90'))
+
+        env.prepend_path('LD_LIBRARY_PATH', self.prefix.lib)
+        env.set('OPAL_PREFIX', self.prefix)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hpcx-mpi/package.py]: This seems to be the same as the run environment. Can wwe factor the changes into a common method that gets called by both `setup_dependent_build_environment` and `setup_run_environment`?

**Review by alalazo — CHANGES_REQUESTED** [review]
Minor request, but otherwise LGTM

File: var/spack/repos/builtin/packages/hpcx-mpi/package.py (hunk 1 above)
Comment (mwkrentel) [reply var/spack/repos/builtin/packages/hpcx-mpi/package.py]: Ok, I'll refactor that.

**Comment by mwkrentel** [comment]
@alalazo Ok, I refactored the build/run environments.
And the ci/gitlab pipeline now says success.
Quick, before the CI changes its mind.  :-)

**Review by alalazo — APPROVED** [review]

**Pull request merged**